lib/styleToFilters.js

```diff
@@ -33,15 +33,15 @@ function styleToFilters(style) {
           if (styleMin < layers[layerName].minzoom) layers[layerName].minzoom = styleMin;
           if (styleMax > layers[layerName].maxzoom) layers[layerName].maxzoom = styleMax;
           // Modify filter
-          if (layers[layerName].filters === true || !style.layers[i].filter) {
+          if (layers[layerName].filters === true || !style.layers[i].filter || isNoOpFilter(style.layers[i].filter)) {
```

Comment:
this would be too aggressive, so for such case:
```
            "filter": [
                "all",
                [
                    "case",
                    ["<=", ["pitch"], 45],
                    true,
                    ["<=", ["distance-from-center"], 1]
                ],
                ["match", ["get", "type"], ["Toll booth"], true, false]
            ],
```
we would want to replace it with:
```
  "filter": [
                "all",
                [
                    "literal",
                     true
                ],
                ["match", ["get", "type"], ["Toll booth"], true, false]
            ],
```
instead of just ['literal', true]